### PR TITLE
Syntax for bytestring based models

### PR DESCRIPTION
**Pull request by nzpr**
This PR unifies syntax for bytestring based models (Validator, StateHash, BlockHash). This is some work from deploymerge branch which belongs to dev.

### CODE REVIEW

File: shared/src/main/scala/coop/rchain/shared/package.scala [hunk 1]
```diff
@@ -18,3 +18,4 @@ trait AllSyntaxShared
     with KeyValueStoreManagerSyntax
     with MonixableSyntax
     with Fs2StreamSyntax
+    with catscontrib.ToBooleanF
```
Comment (zsluedem) [inline shared/src/main/scala/coop/rchain/shared/package.scala:21]: Just curious.
This is not related to byteString, right?

Comment (tgrospic) [reply shared/src/main/scala/coop/rchain/shared/package.scala:21]: I'm ok for ToBooleanF to also be included in this PR. It needs to be renamed.

**Review by zsluedem — APPROVED** [review]
LGTM

**Review by tgrospic — APPROVED** [review]
LGTM

**Comment by nzpr** [comment]
bors merge
